sdk/core/Azure.Core/src/Shared/OperationInternalOfT.cs

```diff
@@ -176,7 +179,7 @@ protected override async ValueTask<Response> UpdateStateAsync(bool async, Cancel
             {
                 if (state.HasSucceeded)
                 {
```

Comment:
debug.Assert? here and in other places where we use `!`

Reply:
Fixed